bliss/predict.py

```diff
@@ -33,8 +36,46 @@ def predict_sdss(cfg):
     return est_cat, images[0], background[0], sdss_plocs, pred
 
 
-def decal_plocs_from_sdss(cfg):
-    """Use wcs to match sdss image with corresponding DeCals catalog for futher plotting."""
+def decals_plocs_from_decals(decals: DarkEnergyCameraLegacySurvey, cfg):
+    # get ra_lim, dec_lim corresponding to DECaLS image(s)
+    width_in_deg = DarkEnergyCameraLegacySurvey.pix_to_deg(cfg.predict.dataset.width)
+    height_in_deg = DarkEnergyCameraLegacySurvey.pix_to_deg(cfg.predict.dataset.height)
+    ra_lim = (
+        cfg.predict.dataset.ra - width_in_deg / 2,
+        cfg.predict.dataset.ra + width_in_deg / 2,
+    )
+    dec_lim = (
+        cfg.predict.dataset.dec - height_in_deg / 2,
+        cfg.predict.dataset.dec + height_in_deg / 2,
+    )
+    tractor_filename = cfg.paths.decals + f"/tractor-{decals.get_brick_name()}.fits"
+    return DecalsFullCatalog.from_file(tractor_filename, ra_lim, dec_lim).plocs[0]
+
+
+def predict_decals(cfg):
+    decals = instantiate(cfg.predict.dataset)
+    assert isinstance(
+        decals, DarkEnergyCameraLegacySurvey
+    ), "decals expected to be a DarkEnergyCameraLegacySurvey, but got {}".format(type(decals))
+    decals_plocs = decals_plocs_from_decals(decals, cfg)
+
+    encoder = instantiate(cfg.encoder).to(cfg.predict.device)
+    enc_state_dict = torch.load(cfg.predict.weight_save_path)
+    encoder.load_state_dict(enc_state_dict)
+    encoder.eval()
+    trainer = instantiate(cfg.predict.trainer)
+    est_cat, images, background, pred = trainer.predict(encoder, datamodule=decals)[0].values()
+    if cfg.predict.plot.show_plot and (decals_plocs is not None):
+        ptc = cfg.encoder.tiles_to_crop * cfg.encoder.tile_slen
+        cropped_image = images[0, 0, ptc:-ptc, ptc:-ptc]
+        cropped_background = background[0, 0, ptc:-ptc, ptc:-ptc]
+        plot_predict(cfg, cropped_image, cropped_background, decals_plocs, est_cat)
+
+    return est_cat, images[0], background[0], decals_plocs, pred
+
+
+def decals_plocs_from_sdss(cfg):
```

Comment:
same here -- it'd be wonderful to avoid any sdss- or decals-specific code (and variable names) in `predict.py`. Just have functions that operate on pairs of generic catalogs and/or datasets.

Reply:
Tracked by #853.